### PR TITLE
[Syntax] Abolish 'backtick' trivia

### PR DESCRIPTION
**Pull request by rintaro**
- Stop producing 'backtick' trivia for escaped identifier token. `` ` ``s  are now parts of the token text
- Adjust and simplify C++ libSyntax APIs
- Remove backtick trivia declaration

rdar://problem/54810608

swift-syntax change: https://github.com/apple/swift-syntax/pull/142

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/142
@swift-ci Please test

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/142
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please Python Lint

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/142
@swift-ci Please test

File: utils/gyb_syntax_support/Trivia.py [hunk 1]
```diff
@@ -8,15 +8,17 @@ def __init__(self, name, comment, serialization_code, characters=[],
         self.name = name
         self.comment = comment
         self.serialization_code = serialization_code
-        self.characters = characters
+        self.characters = tuple(characters)
```
Comment (rintaro) [inline utils/gyb_syntax_support/Trivia.py]: `tuple()` is needed to make it "hashable".

**Review by nkcsgexi — APPROVED** [review]

**Pull request merged**